core/src/main/java/org/kiwix/kiwixmobile/core/webserver/ZimHostActivity.kt

```diff
@@ -0,0 +1,290 @@
+  }
+
+  private fun select(bookOnDisk: BooksOnDiskListItem.BookOnDisk) {
+    val booksList: ArrayList<BooksOnDiskListItem> = ArrayList()
+    booksAdapter.items.map {
+      if (it == bookOnDisk) {
+        it.isSelected = !it.isSelected
+      }
+      booksList.add(it)
+    }
+    booksAdapter.items = booksList
+    saveHostedBooks(booksList)
+  }
+
+  override fun onStart() {
+    super.onStart()
+    bindService()
+  }
+
+  override fun onStop() {
+    super.onStop()
+    unbindService()
+  }
+
+  private fun bindService() {
+    bindService(
+      Intent(this, HotspotService::class.java), serviceConnection,
+      Context.BIND_AUTO_CREATE
+    )
+  }
+
+  private fun unbindService() {
+    hotspotService?.let {
+      unbindService(serviceConnection)
+      it.registerCallBack(null)
+    }
+  }
+
+  override fun onResume() {
+    super.onResume()
+    presenter.loadBooks(sharedPreferenceUtil.hostedBooks)
+    if (ServerUtils.isServerStarted) {
+      ip = ServerUtils.getSocketAddress()
+      layoutServerStarted()
+    }
+  }
+
+  private fun saveHostedBooks(booksList: ArrayList<BooksOnDiskListItem>) {
+    sharedPreferenceUtil.hostedBooks = booksList.asSequence()
+      .filter(BooksOnDiskListItem::isSelected)
+      .filterIsInstance<BookOnDisk>()
+      .mapNotNull { it.book.title }
+      .toSet()
+  }
+
+  private fun layoutServerStarted() {
+    serverTextView.text = getString(R.string.server_started_message, ip)
+    startServerButton.text = getString(R.string.stop_server_label)
+    startServerButton.setBackgroundColor(resources.getColor(R.color.stopServer))
+    bookDelegate.selectionMode = SelectionMode.NORMAL
+    booksAdapter.notifyDataSetChanged()
+  }
+
+  private fun layoutServerStopped() {
+    serverTextView.text = getString(R.string.server_textview_default_message)
+    startServerButton.text = getString(R.string.start_server_label)
+    startServerButton.setBackgroundColor(resources.getColor(R.color.greenTick))
+    bookDelegate.selectionMode = SelectionMode.MULTI
+    booksAdapter.notifyDataSetChanged()
+  }
+
+  override fun onDestroy() {
+    super.onDestroy()
+    presenter.detachView()
+  }
+
+  private fun setUpToolbar() {
+    setSupportActionBar(toolbar)
+    supportActionBar!!.title = getString(R.string.menu_host_books)
+    supportActionBar!!.setHomeButtonEnabled(true)
+    supportActionBar!!.setDisplayHomeAsUpEnabled(true)
+
+    toolbar.setNavigationOnClickListener { onBackPressed() }
+  }
+
+  // Advice user to turn on hotspot manually for API<26
+  private fun startHotspotManuallyDialog() {
+
+    alertDialogShower.show(KiwixDialog.StartHotspotManually(),
+      ::launchTetheringSettingsScreen,
+      null,
+      {
+        progressDialog = ProgressDialog.show(
+          this,
+          getString(R.string.progress_dialog_starting_server), "",
+          true
+        )
+        startService(createHotspotIntent(ACTION_CHECK_IP_ADDRESS))
+        Unit
```

Comment:
Unit can be omitted

Reply:
https://github.com/kiwix/kiwix-android/pull/1569/commits/6b918125101e56fe10b9392a5817f23c5bcac93c